packages/backend/src/services/test-run.ts

```diff
@@ -49,16 +49,34 @@ const testRun = async (options: TestRunOptions) => {
     return { executionStep: triggerExecutionStep }
   }
 
+  // Actions may redirect steps, so we keep track here. We want to loop through
+  // _all_ possible actions during a test run, so that we can let people know if
+  // an action was skipped due to redirection.
+  let nextStepId = actionSteps[0]?.id
+
   for (const actionStep of actionSteps) {
-    const { executionStep: actionExecutionStep } = await processAction({
-      flowId: flow.id,
-      stepId: actionStep.id,
-      executionId,
-    })
+    if (actionStep.id !== nextStepId) {
+      if (!nextStepId || actionStep.id === untilStep.id) {
+        throw new Error(
+          'Unable to test step because it was skipped by conditional logic.',
```

Comment:
Discussed with Jackson - we _may_ not want this. The high level idea is that I wanted user to re-submit test data to test each branch, otherwise they might get confused (_e.g. why is branch B triggering when my test data should only satisfy Branch A?_)

However, this doesn't scale for pipes with many branches....

Reply:
Will be discussing this more with the team

Reply:
let's not throw an error, return up to u, then show u frontend